src/lib/containers/table/SynapseTable.tsx

```diff
@@ -168,15 +168,25 @@ export default class SynapseTable extends React.Component<
   shouldComponentUpdate(
     nextProps: QueryWrapperChildProps & SynapseTableProps,
     nextState: Readonly<SynapseTableState>,
-    nextContext: any,
   ): boolean {
-    this.disableResize()
-    return super.shouldComponentUpdate
-      ? super.shouldComponentUpdate(nextProps, nextState, nextContext)
-      : true
+    // ignore isFetching state variables when checking for change in state
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { isFetchingEntityHeaders: oldIsFetchingEntityHeaders, isFetchingEntityVersion: oldIsFetchingEntityVersion, ...oldState } = this.state
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { isFetchingEntityHeaders: newIsFetchingEntityHeaders, isFetchingEntityVersion: newIsFetchingEntityVersion, ...newState } = nextState
+    const isPropsChange = !eq(this.props, nextProps)
+    const isStateChange = !eq(oldState, newState)
+    const shouldComponentUpdate = isPropsChange || isStateChange
+    if (shouldComponentUpdate) {
+      this.disableResize()
+    }
+    return shouldComponentUpdate
   }
   componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps) {
-    this.getEntityHeadersInData(false)
+    // PORTALS-2081: if the data changed, then get the new entity headers
+    if (!eq(prevProps.data, this.props.data)) {
```

Comment:
This single line fixes PORTALS-2081, make sure the data has changed before getting the new entity headers